src/util/permissions.ts

```diff
@@ -1,256 +1,285 @@
 import { cacheHandlers } from "../api/controllers/cache.ts";
-import { Guild, Role } from "../api/structures/mod.ts";
+import { Role } from "../api/structures/mod.ts";
 import { botID } from "../bot.ts";
-import { Permission, Permissions, RawOverwrite } from "../types/mod.ts";
+import { Errors, Permission, Permissions } from "../types/mod.ts";
 
-/** Checks if the member has this permission. If the member is an owner or has admin perms it will always be true. */
-export async function memberIDHasPermission(
+/** Computes the permissions this member has in the given guild */
+export async function computeBasePermissions(
   memberID: string,
   guildID: string,
-  permissions: Permission[],
 ) {
   const guild = await cacheHandlers.get("guilds", guildID);
-  if (!guild) return false;
+  if (!guild) throw Error(Errors.GUILD_NOT_FOUND);
+  // If the memberID is equal to the guild ownerID we don't need to calculate anything so we return ADMINISTRATOR permission
+  if (guild.ownerID === memberID) return "8";
 
-  if (memberID === guild.ownerID) return true;
+  const member = await cacheHandlers.get("members", memberID);
+  if (!member) throw Error(Errors.MEMBER_NOT_FOUND);
 
-  const member = (await cacheHandlers.get("members", memberID))?.guilds.get(
+  let permissions = BigInt(0);
+  // Calculate the role permissions bits, @everyone role is not in memberRoleIDs so we need to pass guildID manualy
+  permissions |= [
+    ...member.guilds.get(guildID)?.roles || [],
     guildID,
-  );
-  if (!member) return false;
-
-  return memberHasPermission(memberID, guild, member.roles, permissions);
-}
-
-/** Checks if the member has this permission. If the member is an owner or has admin perms it will always be true. */
-export function memberHasPermission(
-  memberID: string,
-  guild: Guild,
-  memberRoleIDs: string[],
-  permissions: Permission[],
-) {
-  if (memberID === guild.ownerID) return true;
-
-  const permissionBits = memberRoleIDs.map((id) =>
-    guild.roles.get(id)?.permissions
-  )
-    // Removes any edge case undefined
+  ]
+    .map((id) => guild.roles.get(id)?.permissions)
     .filter((id) => id)
+    // Removes any edge case undefined
     .reduce((bits, perms) => {
       bits |= BigInt(perms);
       return bits;
     }, BigInt(0));
-
-  if (permissionBits & BigInt(Permissions.ADMINISTRATOR)) return true;
-
-  return permissions.every((permission) =>
-    permissionBits & BigInt(Permissions[permission])
-  );
+  // If one role has ADMINISTRATOR permissions we don't need to return the specifiv permissions so we return ADMINISTRATOR permission
+  if (permissions & BigInt(Permissions.ADMINISTRATOR)) return "8";
+  // Return the members permission bits as a string
+  return permissions.toString();
 }
 
-export async function botHasPermission(
-  guildID: string,
-  permissions: Permission[],
+/** Computes the members Channel Overwrites */
```

Comment:
A better description of what the function does would be appreciated.

Reply:
`Computes the permissions this member has in the given Channel`